Dockerfile

```
@@ -443,29 +443,73 @@ RUN \
 
 FROM sdk-libc as sdk-go-prep
 
-ENV GOVER="1.21.9"
+# Set up the environment for building.
+ENV GOOS="linux"
+ENV CGO_ENABLED=1
+ENV CFLAGS="-O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-clash-protection"
+ENV CXXFLAGS="${CFLAGS}"
+ENV LDFLAGS="-Wl,-z,relro -Wl,-z,now"
+ENV CGO_CFLAGS="${CFLAGS}"
+ENV CGO_CXXFLAGS="${CXXFLAGS}"
+ENV CGO_LDFLAGS="${LDFLAGS}"
+
+ENV GO111MODULE="auto"
+
 ENV AWS_LC_FIPS_VER="2.0.9"
 
 USER root
 RUN dnf -y install golang
 
+ENV GOVER_1_21="1.21.9"
+ENV GOVER_1_22="1.22.1"
+
+# =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+FROM sdk-go-prep as sdk-go-1.21-prep
+
 USER builder
-WORKDIR /home/builder/sdk-go
-COPY ./hashes/go /home/builder/hashes
+
+WORKDIR /home/builder/sdk-go-${GOVER_1_21}
+
+COPY ./hashes/go-${GOVER_1_21} /home/builder/hashes
+COPY ./helpers/go/* ./
+COPY patches/go-${GOVER_1_21}/* ./
```

We want to keep the filenames the same so that all we have to do is update the contents and the version in Dockerfile.

```suggestion
COPY ./hashes/go-1.21 /home/builder/hashes
COPY ./helpers/go/* ./
COPY ./patches/go-1.21/* ./
```